### DISCUSSION TITLE
Hello, I want to know how can i create a mbr usb device to boot the tianocore

### DISCUSSION THREAD

**Discussion by 9Bakabaka** [tianocore/edk2#5847] [Q&A]
I want to create a usb device for the tianocore and then boot from the mbr. However, I found that i need to use the "DuetPkg" But i  did not find it in the master branch. Did it have been dropped? Or i need to use another Pkg to create the usb device

**Comment by ajfish** [comment]
> On Jul 2, 2024, at 8:20 AM, 9Bakabaka ***@***.***> wrote:
> 
> 
> I want to create a usb device for the tianocore and then boot from the mbr.
> 
If you want to make a bootable removable media in UEFI you just need to put an UEFI application @ \EFI\BOOT\BOOT{machine type short-name}.EFI.
So for x86_64 that would be:  \EFI\BOOT\BOOTx64.EFI
So you can have MBR with a FAT32 partition and  \EFI\BOOT\BOOTx64.EFI is a file in the FAT32 file system.

Note there is a typo:
\EFIBOOTBOOT{machine type short-name}.EFI

Should be:
\EFI\BOOT\BOOT{machine type short-name}.EFI
https://uefi.org/specs/UEFI/2.10/03_Boot_Manager.html#removable-media-boot-behavior

When an OS installs it writes an UEFI variable to point to the location. There is a registry so vendors can own directories on the EFI System Partition. This prevents collisions when different OS venders install to the UEFI System Partition.
https://uefi.org/registry
